src/components/ColumnAnnotationCard.tsx

```diff
@@ -55,6 +57,12 @@ function ColumnAnnotationCard({
     onStandardizedVariableChange(id, newStandardizedVariable);
   };
 
+  const isDataTypeDisabled =
```

Comment:
How about

```suggestion
  const isDataTypePredefined =
```
Because "disabled" is the UI concern, not the internal logic concern. i.e. we know and store the data type.